### PR TITLE
new: add support for `delete_module` syscall

### PR DESCRIPTION
**Pull request by loresuso**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

/kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area API-version

> /area build

> /area CI

/area driver-kmod

/area driver-bpf

/area driver-modern-bpf

> /area libscap-engine-bpf

> /area libscap-engine-gvisor

> /area libscap-engine-kmod

> /area libscap-engine-modern-bpf

> /area libscap-engine-nodriver

> /area libscap-engine-noop

> /area libscap-engine-source-plugin

> /area libscap-engine-savefile

> /area libscap

> /area libpman

> /area libsinsp

/area tests

> /area proposals

**Does this PR require a change in the driver versions?**

> /version driver-API-version-major

> /version driver-API-version-minor

> /version driver-API-version-patch

> /version driver-SCHEMA-version-major

> /version driver-SCHEMA-version-minor

> /version driver-SCHEMA-version-patch

**What this PR does / why we need it**:

Add support for `delete_module` syscall. This could be useful to write rules, especially on the `name` parameter to understand which kernel module was unloaded (instead of the current generic event with no info about this).

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: test/drivers/test_suites/syscall_exit_suite/delete_module_x.cpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+	auto evt_test = get_syscall_event_test(__NR_delete_module, EXIT_EVENT);
+	evt_test->enable_capture();
+
+	/*=============================== TRIGGER SYSCALL  ===========================*/
+
+	/*
+	 * Call the `delete_module` syscall
+	 */
+	assert_syscall_state(SYSCALL_FAILURE, "delete_module", syscall(__NR_delete_module, module_name, 0));
```
Comment (FedeDP) [inline test/drivers/test_suites/syscall_exit_suite/delete_module_x.cpp:18]: Can we test with flags != 0 here?

Comment (loresuso) [reply test/drivers/test_suites/syscall_exit_suite/delete_module_x.cpp:18]: Of course, I'm gonna add it!

File: driver/flags_table.c [hunk 2]
```diff
@@ -755,4 +755,10 @@ const struct ppm_name_value bpf_commands[] = {
 	{"BPF_LINK_DETACH", PPM_BPF_LINK_DETACH},
 	{"BPF_PROG_BIND_MAP", PPM_BPF_PROG_BIND_MAP},
 	{0,0},
+};
+
+const struct ppm_name_value delete_module_flags[] = {
+	{"O_NONBLOCK", PPM_O_NONBLOCK},
+	{"O_TRUNC", PPM_O_TRUNC},
```
[the remark this answers is not in the document]
Comment (loresuso) [reply driver/flags_table.c:762]: Thanks!

File: driver/SCHEMA_VERSION [hunk 3]
```diff
@@ -1 +1 @@
-2.17.1
```
Comment (leogr) [inline driver/SCHEMA_VERSION:1]: :+1:

File: driver/SCHEMA_VERSION [hunk 4]
```diff
@@ -1 +1 @@
...
+2.18.0
```
Comment (leogr) [inline driver/SCHEMA_VERSION:1]: :+1:

**Review by FedeDP — APPROVED** [review]
/approve

**Review by Andreagit97 — APPROVED** [review]
/approve

**Pull request merged**